api/v1alpha1/httproute_types.go

```diff
@@ -227,12 +228,31 @@ type HTTPRouteAction struct {
 	Extension *RouteActionExtensionObjectReference `json:"extension" protobuf:"bytes,2,opt,name=extension"`
 }
 
+// TargetPort specifies the destination port name or number of a target.
+type TargetPort intstr.IntOrString
```

Comment:
In Ingress we're moving to a struct that includes `name` and `string` references to avoid using the `intstr` type and all that comes along with it. Would that make sense here as well?

Reply:
I'd actually be in favor of simply targeting a port number here instead, assuming we don't need to target a port name on a Service.